src/System.Net.HttpListener/src/System/Net/Managed/WebSockets/WebSocketValidate.Http.Managed.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace System.Net.WebSockets
+{
+    internal static partial class WebSocketValidate
+    {
+        internal static async Task<HttpListenerWebSocketContext> AcceptWebSocketAsyncCore(HttpListenerContext context,
```

Comment:
Why is all of this functionality in a type called WebSocketValidate?  It looks like it's the bulk of the implementation for actually accepting a web sockets connection, whereas the WebSocketValidate class was intended to be for validation, e.g. checking status and throwing exceptions, checking buffer sizes and throwing exceptions, etc.